packages/gestalt/src/utils/positioningUtils.js

```diff
@@ -135,24 +137,24 @@ export function getPopoverDir({
 }: {|
   popoverSize: Dimensions,
   idealDirection: PopoverDir,
-  triggerRect: ClientRect,
+  triggerRect: ?ClientRect,
   windowSize: Window,
   isScrollBoundaryContainer?: boolean,
 |}): MainDirections {
   // Calculates the available space if we were to place the popover in the 4 main directions
   // to determine which 'quadrant' to position the popover inside of
-  let up = triggerRect.top - popoverSize.height - CARET_HEIGHT;
-  let right = windowSize.width - popoverSize.width - CARET_HEIGHT - triggerRect.right;
-  let down = windowSize.height - popoverSize.height - CARET_HEIGHT - triggerRect.bottom;
-  let left = triggerRect.left - popoverSize.width - CARET_HEIGHT;
+  let up = (triggerRect?.top ?? 0) - popoverSize.height - CARET_HEIGHT;
```

Comment:
Should / could we do the fallback for `top` / `right` / `bottom` / `left` to `0` whenever we initialize `triggerRect`? If we do that though, make sure we have good tests in place so we don't introduce any bugs.

Reply:
Possibly — I put minimal thought into this as it's all related to Popover, which there's a pretty good chance we replace with a third party library sometime soon-ish. It'll be really nice to offload that responsibility and delete all this code